store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreConfiguredGraphIdsTest.java

```diff
@@ -0,0 +1,299 @@
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreTestUtil.STRING;
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreTestUtil.contextTestUser;
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreTestUtil.loadFederatedStoreFrom;
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreTestUtil.loadStoreProperties;
+import static uk.gov.gchq.gaffer.store.TestTypes.DIRECTED_EITHER;
+import static uk.gov.gchq.gaffer.user.StoreUser.testUser;
+
+public class FederatedStoreConfiguredGraphIdsTest {
+
+    private static final Schema STRING_SCHEMA = new Schema.Builder()
+            .type(STRING, new TypeDefinition.Builder()
+                    .clazz(String.class)
+                    .serialiser(new StringSerialiser())
+                    .aggregateFunction(new StringConcat())
+                    .build())
+            .build();
+
+    private static final StoreProperties PROPERTIES = loadStoreProperties("properties/singleUseMapStore.properties");
+    private static final String STORE_PROP_ID = "storeProp";
+    private static final String GRAPH_ID_SELECTED = "selectedGraphId";
+    private static final String SCHEMA_SELECTED = "selectedSchema";
+    private static final String GRAPH_ID_UNSELECTED = "unselectedGraphId";
+    private static final String SCHEMA_UNSELECTED = "unselectedSchema";
+    public static final String FEDERATED_STORE_CONFIGURED_GRAPH_IDS_PROPERTIES = "configuredProperties/federatedStoreConfiguredGraphIds.properties";
+    public static final String FEDERATED_STORE_CONFIGURED_GRAPH_IDS_JSON = "configuredProperties/federatedStoreConfiguredGraphIds.json";
+    public static final String FEDERATED_STORE_CONFIGURED_GRAPH_IDS_FAIL_PATH_MISSING_PROPERTIES = "configuredProperties/federatedStoreConfiguredGraphIdsFailPathMissing.properties";
+    public static final String FEDERATED_STORE_CONFIGURED_GRAPH_IDS_FAIL_FILE_INVALID_PROPERTIES = "configuredProperties/federatedStoreConfiguredGraphIdsFailFileInvalid.properties";
+
+    @Test
+    void shouldFailConfiguredGraphIdsPathMissing() {
+        assertThatExceptionOfType(IllegalArgumentException.class)
+                .isThrownBy(() -> getFederatedStoreFromProperties(FEDERATED_STORE_CONFIGURED_GRAPH_IDS_FAIL_PATH_MISSING_PROPERTIES))
+                .withMessageContaining("Failed to create input stream for path: fail");
+    }
+
+    @Test
+    void shouldFailConfiguredGraphIdsFileInvalid() {
+        assertThatExceptionOfType(IllegalArgumentException.class)
+                .isThrownBy(() -> getFederatedStoreFromProperties(FEDERATED_STORE_CONFIGURED_GRAPH_IDS_FAIL_FILE_INVALID_PROPERTIES))
+                .withMessageContaining("Could not initialise the store with provided arguments.")
+                .withStackTraceContaining("Cannot deserialize value of type `java.util.ArrayList<java.lang.Object>` from String value");
+    }
+
+    @Test
+    void shouldReturnSelectedGraphIdByDefaultFromJsonConfig()
+            throws Exception {
```

Comment:
Why is the throw on another line? It looks odd and there's plenty of space for it in the usual place.